File: src/cn/vove7/plugin/rest/filetype/RestFileTypeFactory.java
package cn.vove7.plugin.rest.filetype;

import com.intellij.openapi.fileTypes.FileTypeConsumer;
import com.intellij.openapi.fileTypes.FileTypeFactory;

import org.jetbrains.annotations.NotNull;

/**
 *
 */
public class RestFileTypeFactory extends FileTypeFactory {
    @Override
    public void createFileTypes(@NotNull FileTypeConsumer consumer) {
        consumer.consume(RestFileType.INSTANCE, "rest");
    }
}

File: src/main/kotlin/cn/vove7/plugin/rest/filetype/RestFileTypeFactory.kt
package cn.vove7.plugin.rest.filetype

import com.intellij.openapi.fileTypes.FileTypeFactory
import com.intellij.openapi.fileTypes.FileTypeConsumer
import cn.vove7.plugin.rest.filetype.RestFileType

/**
 *
 */
class RestFileTypeFactory : FileTypeFactory() {
    override fun createFileTypes(consumer: FileTypeConsumer) {
        consumer.consume(RestFileType.INSTANCE, "rest")
    }
}